src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -1548,16 +1549,15 @@ XclBin::setKeyValue(const std::string& _keyValue)
     }
 
     if (sKey == "action_mask") {
-      std::vector<std::string> masks;
-      boost::split(masks, sValue, boost::is_any_of("|"));
-      m_xclBinHeader.m_header.m_actionMask = 0;
-      for (const auto& mask : masks) {
-        if (mask == "LOAD_AIE") {
-          m_xclBinHeader.m_header.m_actionMask |= AM_LOAD_AIE;
-        } else {
-          auto errMsg = boost::format("ERROR: Unknown bit mask '%s' for the key '%s'. Key-value pair: '%s'.") % mask % sKey % _keyValue;
-          throw std::runtime_error(errMsg.str());
-        }
+      if (sValue == "LOAD_AIE") {
+        m_xclBinHeader.m_header.m_actionMask |= AM_LOAD_AIE;
+      }
+      else if (sValue == "LOAD_PDI") {
+        m_xclBinHeader.m_header.m_actionMask |= AM_LOAD_PDI; 	
+      }
+      else {
+        auto errMsg = boost::format("ERROR: Unknown bit mask '%s' for the key '%s'. Key-value pair: '%s'.") % sValue % sKey % _keyValue;
+        throw std::runtime_error(errMsg.str());
```

Comment:
The old code seems to support "mask1|mask2|...", but the new code assumes LOAD_AIE and LOAD_PDI are exclusive, have you confirmed that it is the expected behavior?

Reply:
The old code doesn't recognize command line option like SYS:action_mask:LOAD_AIE|LOAD_PDI. The developer didn't have any processing logic to recognize multiple masks in one key value pair.  The error when I got when I tried this  SYS:action_mask:LOAD_AIE|LOAD_PDI option was like "LOAD_PDI command not found" which makes sense as it considers '|' char as pipe symbol in linux command prompt. At the same time the new code does support multiple masks like LOAD_AIE and LOAD_PDI  together as this if-else logic iteratively sets the action mask for each key-value pair i,e. once for SYS:action_mask:LOAD_AIE and another time for SYS:action_mask:LOAD_PDI.

Reply:
what would the command line look like if user wants to specify both masks?

Reply:
Something like this "xclbinutil --key-value SYS:action_mask:LOAD_AIE SYS:action_mask:LOAD_PDI --output final.xclbin"

Output of --info on final.xclbin:

xclbin Information
------------------
   Generated by:           <unknown>
   Version:                2.18.0
   Kernels:                <unknown>
   Signature:              
   **Action Mask(s):**         LOAD_AIE LOAD_PDI 
   Content:                
   UUID (xclbin):          94b5e2a1-a0bc-bf32-cdb7-82b3fe9ddc2b
   Sections: